src/main/java/FrameworksDrivers/MainPageView.java

```diff
@@ -142,10 +142,14 @@ public void actionPerformed(ActionEvent e) {
     }
     public void updatePage(Object[] info) {
         //Just switch the user
-        if(info != null){
+        if(info == null){
```

Comment:
If no displayed user is given(ie coming from a different view) the presenter reruns which will give a display user